src/Sentry/Internal/MemoryInfo.cs

```diff
@@ -81,23 +82,24 @@ internal sealed class MemoryInfo : IJsonSerializable
 #endif
         public void WriteTo(Utf8JsonWriter writer, IDiagnosticLogger? logger)
         {
+            //WriteNumberIfNotZero since on OS that dont implement all props, those props are stubbed out to zero
             writer.WriteStartObject();
 
-            writer.WriteNumber("allocated_bytes", AllocatedBytes);
-            writer.WriteNumber("fragmented_bytes", FragmentedBytes);
-            writer.WriteNumber("heap_size_bytes", HeapSizeBytes);
-            writer.WriteNumber("high_memory_load_threshold_bytes", HighMemoryLoadThresholdBytes);
-            writer.WriteNumber("total_available_memory_bytes", TotalAvailableMemoryBytes);
-            writer.WriteNumber("memory_load_bytes", MemoryLoadBytes);
+            writer.WriteNumberIfNotZero("allocated_bytes", AllocatedBytes);
+            writer.WriteNumberIfNotZero("fragmented_bytes", FragmentedBytes);
+            writer.WriteNumberIfNotZero("heap_size_bytes", HeapSizeBytes);
+            writer.WriteNumberIfNotZero("high_memory_load_threshold_bytes", HighMemoryLoadThresholdBytes);
+            writer.WriteNumberIfNotZero("total_available_memory_bytes", TotalAvailableMemoryBytes);
+            writer.WriteNumberIfNotZero("memory_load_bytes", MemoryLoadBytes);
 
 #if NET5_0_OR_GREATER
-            writer.WriteNumber("total_committed_bytes", TotalCommittedBytes);
-            writer.WriteNumber("promoted_bytes", PromotedBytes);
-            writer.WriteNumber("pinned_objects_count", PinnedObjectsCount);
-            writer.WriteNumber("pause_time_percentage", PauseTimePercentage);
-            writer.WriteNumber("index", Index);
-            writer.WriteNumber("generation", Generation);
-            writer.WriteNumber("finalization_pending_count", FinalizationPendingCount);
+            writer.WriteNumberIfNotZero("total_committed_bytes", TotalCommittedBytes);
+            writer.WriteNumberIfNotZero("promoted_bytes", PromotedBytes);
+            writer.WriteNumberIfNotZero("pinned_objects_count", PinnedObjectsCount);
+            writer.WriteNumberIfNotZero("pause_time_percentage", PauseTimePercentage);
+            writer.WriteNumberIfNotZero("index", Index);
+            writer.WriteNumberIfNotZero("generation", Generation);
+            writer.WriteNumberIfNotZero("finalization_pending_count", FinalizationPendingCount);
```

Comment:
Would 0 be a valid value here? https://docs.microsoft.com/en-us/dotnet/api/system.gcmemoryinfo.finalizationpendingcount?view=net-6.0#system-gcmemoryinfo-finalizationpendingcount